src/Datepicker/DatesList.tsx

```diff
@@ -1,29 +1,46 @@
 import React, { FC } from 'react'
 import styled from 'styled-components'
+import { getISODay } from 'date-fns'
 
 import { theme } from '../theme'
 
 import { Day } from './types'
 
 type Props = {
   items: Day[]
+  firstDayShift: boolean
   handleDateSelect: (date: Date) => void
 }
 
-export const DatesList: FC<Props> = ({ items, handleDateSelect }) => (
-  <Container>
-    {items.map((item: Day, i) => (
-      <ListItem
-        key={i}
-        disabled={item.disabled}
-        className={`ListItem ${item.active ? 'active' : ''}`}
-        onClick={() => handleDateSelect(item.date)}
-      >
-        {item.label}
-      </ListItem>
-    ))}
-  </Container>
-)
+const getBlankDaysCount = (firstDayOfTheMonth: Date) => {
+  const dayOfTheWeek = getISODay(firstDayOfTheMonth)
+  return dayOfTheWeek - 1
+}
+
+export const DatesList: FC<Props> = ({
+  items,
+  handleDateSelect,
+  firstDayShift,
+}) => {
+  return (
+    <Container>
+      {firstDayShift &&
+        Array(getBlankDaysCount(items[0].date))
+          .fill(null)
+          .map((_, index) => <ListItem key={`blankItem-${index}`} disabled />)}
```

Comment:
I moved the blank day (calendar display mode) inside the list, I don't think it made sense to be inside the day generation function as it's not related to the data itself, only presentational